main.rb

```diff
@@ -0,0 +1,175 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    y = 0
+    arr = to_a
+    while y < arr.length
+      yield arr[y]
+      y += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    y = 0
+    arr = to_a
+    while y < arr.length
+      yield arr[y], y
+      y += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    access = []
+    my_each do |val|
+      access << val if yield val
+    end
+    access
+  end
+
+  def my_all?(par = nil)
+    return test_parm(par) unless par.nil?
+
+    return (my_all? { |x| !x.nil? && x != false }) unless block_given?
+
+    my_each do |a|
+      return false unless yield a
+    end
+    true
+  end
+
+  def test_parm(par)
+    if par.class == Class
+      my_all? { |x| x.class == par }
+    elsif par.class == Regexp
+      my_all? { |x| x =~ par }
+    else
+      my_all? { |x| x == par }
+    end
+  end
+
+  def my_any?(par = nil)
+    return test_param_any(par) unless par.nil?
+
+    return (my_any? { |x| !x.nil? && x != false }) unless block_given?
+
+    my_each do |a|
+      return true if yield a
+    end
+    false
+  end
+
+  def test_param_any(par)
+    if par.class == Class
+      my_any? { |x| x.class == par }
+    elsif par.class == Regexp
+      my_any? { |x| x =~ par }
+    else
+      my_any? { |x| x == par }
+    end
+  end
+
+  def my_none?(par = nil)
+    return test_par_none(par) unless par.nil?
+
+    return (my_none? { |x| !x.nil? && x != false }) unless block_given?
+
+    my_each do |a|
+      return false if yield a
+    end
+    true
+  end
+
+  def test_par_none(par)
+    if par.class == Class
+      my_none? { |x| x.class == par }
+    elsif par.class == Regexp
+      my_none? { |x| x =~ par }
+    else
+      my_none? { |x| x == par }
+    end
+  end
+
+  def my_count(par = nil)
+    cnt = 0
+    return my_count { |x| x == par } unless par.nil?
+    return (my_count { |_x| true }) unless block_given?
+
+    my_each do |a|
+      cnt += 1 if yield a
+    end
+    cnt
+  end
+
+  def my_map
+    return to_enum unless block_given?
+
+    new_array = []
+    my_each do |value|
+      new_array << (yield value)
+    end
+    new_array
+  end
+
+  def my_inject(nval = nil, nsym = nil, nproc = nil)
+    temp_arr = to_a
+    params = compare_params([temp_arr[0], :+, proc {}], [nval, nsym, nproc])
+    total = nil
+    temp_arr.unshift(params[0]) unless params[0].nil?
+    return symbol_inject(params[1], temp_arr) unless params[1].nil?
+
+    temp_arr.my_each_with_index do |value, index|
+      total = if index.zero?
+                value
+              elsif params[2].nil?
+                yield total, value
+              else
+                params[2].call(total, value)
+              end
+    end
+    total
+  end
```

Comment:
- [ ] modify your **my_inject** so it takes symbols or strings as parameters

For example: `[1,2,3].my_inject(:+) `should equal `[1,2,3].inject(:+)`

HINT: Investigate the [send](https://ruby-doc.org/core-2.6.3/Object.html#method-i-send) method. Since `2.send(:+, 2) == 4`